File: 1.1/src/cz/mg/vulkan/PFNvkGetValidationCacheDataEXT.java
package cz.mg.vulkan;

public class PFNvkGetValidationCacheDataEXT extends VkFunctionPointer {
    public PFNvkGetValidationCacheDataEXT() {
    }

    public PFNvkGetValidationCacheDataEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkGetValidationCacheDataEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetValidationCacheDataEXT(long value) {
        setValue(value);
    }

    public PFNvkGetValidationCacheDataEXT(VkInstance instance) {
        super(instance, new VkString("vkGetValidationCacheDataEXT"));
    }

    public void call(VkDevice device, VkValidationCacheEXT validationCache, VkSize pDataSize, VkObject pData, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), validationCache != null ? validationCache.getVkAddress() : VkPointer.getNullAddressNative(), pDataSize != null ? pDataSize.getVkAddress() : VkPointer.NULL, pData != null ? pData.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long device, long validationCache, long pDataSize, long pData, long rval);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetValidationCacheDataEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkGetValidationCacheDataEXT_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong validationCache, jlong pDataSize, jlong pData){
    (void)env;
    (void)jc;
    PFN_vkGetValidationCacheDataEXT f = (PFN_vkGetValidationCacheDataEXT)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkValidationCacheEXT*)jniLongToPointer(validationCache)),
        ((size_t*)jniLongToPointer(pDataSize)),
        ((void*)jniLongToPointer(pData))
    );
}

